Sources/Support/StoreMessagesHelper.swift

```diff
@@ -0,0 +1,191 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreMessagesHelper.swift
+//
+//  Created by Antonio Rico Diez on 27/9/23.
+
+import StoreKit
+
+class StoreMessagesHelper {
+
+    private let systemInfo: SystemInfo
+    private let showStoreMessagesAutomatically: Bool
+    private let storeMessagesProvider: StoreMessagesProvider
+
+    private var deferredMessages: [StoreMessage] = []
+
+    init(systemInfo: SystemInfo,
+         showStoreMessagesAutomatically: Bool) {
+        self.systemInfo = systemInfo
+        self.showStoreMessagesAutomatically = showStoreMessagesAutomatically
+        self.storeMessagesProvider = StoreMessagesProviderWrapper()
+    }
+
+    @available(iOS 16.0, *)
+    init(systemInfo: SystemInfo,
+         showStoreMessagesAutomatically: Bool,
+         storeMessagesProvider: StoreMessagesProvider = StoreMessagesProviderWrapper()) {
+        self.systemInfo = systemInfo
+        self.showStoreMessagesAutomatically = showStoreMessagesAutomatically
+        self.storeMessagesProvider = storeMessagesProvider
+    }
+
+    #if os(iOS)
+
+    @available(iOS 16.4, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    func deferMessagesIfNeeded() async throws {
+        guard !self.showStoreMessagesAutomatically else {
+            return
+        }
+        Task(priority: .background) { [weak self] in
+            guard let storeMessagesProvider = self?.storeMessagesProvider else {
+                return
+            }
+            for try await message in storeMessagesProvider.messages {
+                await MainActor.run { [weak self] in
+                    self?.deferredMessages.append(message)
+                }
+            }
+        }
+    }
+
+    @available(iOS 16.4, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    @MainActor
+    func showStoreMessages(types: Set<StoreMessageType>) {
+        for message in self.deferredMessages {
+            if let messageType = message.reason.messageType {
+                if types.contains(messageType) {
```

Comment:
you can combine these ifs into one with a comma like 
```suggestion
            if let messageType = message.reason.messageType,
               if types.contains(messageType) {
```